src/components/controls/InlineEditableControl.js

```diff
@@ -210,7 +212,20 @@ export default class InlineEditableControl extends React.Component {
     if (prevState.previousValue !== this.state.currentValue
         && prevState.editable === true
         && this.state.editable === false) {
-      this.props.onChange(this.state.currentValue);
+      if (this.props.onChangeConfirm) {
+        this.props.onChangeConfirm(this.state.currentValue)
+        .then((success) => {
+          if (!success) {
+            this.setState({
```

Comment:
I understand what you're trying to do here, but I think we can improve on the user experience. As it stands right now, your code will just revert back to the old value without any indication of error. A better approach might be to leave the input as it is, in other words, leave it in edit mode and don't change the value, and instead change the border color to be red or something to indicate an error, perhaps even an error message (but that might be out of scope for now).